tests/foreman/virtwho/cli/test_esx_sca.py

```diff
@@ -502,3 +505,93 @@ def test_positive_remove_env_option(
+        name_1 = form_data_cli['name']
+        virtwho_config_cli_1 = target_sat.cli.VirtWhoConfig.create(form_data_cli)[
+            'general-information'
+        ]
+        config_file_1 = get_configure_file(virtwho_config_cli_1['id'])
+        command = get_configure_command(virtwho_config_cli_1['id'], module_sca_manifest_org.name)
+        deploy_configure_by_command(
+            command, form_data_cli['hypervisor-type'], org=module_sca_manifest_org.label
+        )
+        rhsm_username_1 = get_configure_option('rhsm_username', config_file_1)
+
+        # create the second virt-who config and get the service account for the first virt-who config on module_sca_manifest_org
+        name_2 = gen_string('alpha')
+        form_data_cli['name'] = name_2
+        virtwho_config_cli_2 = target_sat.cli.VirtWhoConfig.create(form_data_cli)[
+            'general-information'
+        ]
+        config_file_2 = get_configure_file(virtwho_config_cli_2['id'])
+        command_2 = get_configure_command(virtwho_config_cli_2['id'], module_sca_manifest_org.name)
+        deploy_configure_by_command(
+            command_2, form_data_cli['hypervisor-type'], org=module_sca_manifest_org.label
+        )
+        rhsm_username_2 = get_configure_option('rhsm_username', config_file_2)
+        # verify the two service account belong to the same org
+        assert rhsm_username_1 == rhsm_username_2
+        # Create a different org and then create virt-who config
+        ORG_DATA = {'name': f'virtwho_fake_{gen_string("alpha")}'}
+        org = target_sat.api.Organization(name=ORG_DATA['name']).create()
+        target_sat.api.Location(organization=[org]).create()
+        form_data_cli['organization-id'] = org.id
+        form_data_cli['name'] = gen_string('alpha')
+        virtwho_config_cli_3 = target_sat.cli.VirtWhoConfig.create(form_data_cli)[
+            'general-information'
+        ]
+        command = get_configure_command(virtwho_config_cli_3['id'], org.name)
+        deploy_configure_by_command(
+            command, form_data_cli['hypervisor-type'], debug=True, org=org.label
+        )
+        config_file_3 = get_configure_file(virtwho_config_cli_3['id'])
+        rhsm_username_3 = get_configure_option('rhsm_username', config_file_3)
+        assert rhsm_username_3 != rhsm_username_1
+        # hypervisors and guest mapping info send by group of org and service account exist in rhsm.log
+        service_account_message = [
+            f"Authenticating with RHSM username {rhsm_username_1}",
+            f"Authenticating with RHSM username {rhsm_username_3}",
+            f"Host-to-guest mapping being sent to '{module_sca_manifest_org}'",
+            f"Host-to-guest mapping being sent to '{org}'",
+        ]
+        for item in service_account_message:
+            check_message_in_rhsm_log(item)
+        # delete one of the virt-who config on module_sca_manifest_org, verify service account  rhsm_username_1 exit
```

Comment:
Again, possibly code repeating anti-pattern. Also, please, proof-read those comments.

Reply:
@lhellebr which code is code repeating anti-pattern?  will update the comments

Reply:
In these two virt-who config deletions, everything but the target and assertion is the same. I suggest rewriting this to avoid repeating.

Reply:
> In these two virt-who config deletions, everything but the target and assertion is the same. I suggest rewriting this to avoid repeating.

@lhellebr Have submit the update